src/network/networkd-network.c

```diff
@@ -97,6 +97,25 @@ void network_apply_anonymize_if_set(Network *network) {
         network->dhcp_use_timezone = false;
 }
 
+static void network_set_stacked_netdevs_mtu(Network *network) {
+        uint32_t mtu = 0;
+        NetDev *dev;
+        Iterator i;
+
+        HASHMAP_FOREACH(dev, network->stacked_netdevs, i) {
+                if (dev->kind == NETDEV_KIND_VLAN && dev->mtu > 0 && dev->mtu + 4 > mtu)
+                        /* See vlan_dev_change_mtu() in kernel.
+                         * Note that the additional 4bytes may not be necessary for all devices. */
+                        mtu = dev->mtu + 4;
```

Comment:
I'd drop the `&& dev->mtu + 4 > mtu` part above, and simply say `mtu = MAX(mtu, dev->mtu + 4)` here, less repeats.